### DISCUSSION TITLE
Ability to turn off live search?

### DISCUSSION THREAD

**Discussion by peterpanocha** [quodlibet/quodlibet#4590] [Ideas]
Would it be possible to add the ability for a user to turn off "live searching" as you type in the search field and only begin the search when user hits "Enter?"

Just migrated to this awesome program and wanna thank y'all by the way!

**Comment by Meriipu** [comment]
Not sure if it is possible currently but it sounds like a great idea

**Reply by peterpanocha** [reply]
Try going into the config file and changing "eager.search" to "false"

**Comment by declension** [comment]
Somehow I couldn't remember or even easily find how to do this, but here goes:

* Right click the search bar.
* (de)select _Search after typing_ (:warning: I think there's a display bug here the first time, but can't confirm right now)
* continue to type
* note how no results are updated until you press enter etc

FWIW (e.g. if you prefer editing QL config by hand), this is persisted as `eager_search`

**Reply by peterpanocha** [reply]
Yes thank you! The "Search After Typing" setting did not solve the issue, but going into the config file and turning "eager.search" to "false" did the trick!

**Reply by declension** [reply]
(you need to toggle it twice I think, it's a bug on the UI side)

**Reply by casdeiro** [reply]
I'm trying to get rid of this search-while-you-type thing because it causes QL to go unresponsive in the Cover Grid. But I don't have such a variable 'eager.search' in my .config/quodlibet/config file. Is it somewhere else? Can I just add it with the false value?

**Reply by casdeiro** [reply]
> (you need to toggle it twice I think, it's a bug on the UI side)

It doesn't work for me, even toggling it twice. It keeps on freezing after 2 or 3 characters are typed, so I guess it keeps on trying to do the search as I type.

**Reply by declension** [reply]
The config is `settings.eager_search`, so the config file should read:

```
[settings]
# Other stuff...
eager_search = false
```

**Reply by casdeiro** [reply]
Thanks, Declension. I had put it as "false", but somehow the program put it back to "true", as I've discovered. Let's hope it doesn't get changed again. Maybe related to the bug you mentioned before.

**Reply by casdeiro** [reply]
Confirmed: the program overwrites (don't know exactly when or why) my 'false' value with 'true' :-( 
This is a very annoying bug or what?

**Reply by casdeiro** [reply]
Nobody answered this. It's so annoying that I'm thinking about giving up QuodLibet. Is there a way to put 

`eager_search = false`

so that QuodLibet doesn't overwrite it of its own??

I can't get rid of the f*ck*ng "eager_search"!!!

**Reply by casdeiro** [reply]
Changing
`"eager_search": "true",`
line at /usr/lib/python3/dist-packages/quodlibet/**config.py**
to
`"eager_search": "false",`

seems to have fixed the problem.

**Reply by casdeiro** [reply]
I'm afraid I hurried up a little in saying it was a solution! :-/

Next time I opened QuodLibet, eager_search is again set to true!

So I guess there's another piece of code overwriting it again and again... :-(

Any clues, anybody?

**Reply by casdeiro** [reply]
Tried now:

At /usr/lib/python3/dist-packages/quodlibet/qltk/searchbar.py
changed:
```
        cb = ConfigCheckMenuItem(
            _("Search after _typing"), 'settings', 'eager_search',
            populate=True)
```
To
```
        cb = ConfigCheckMenuItem(
            _("Search after _typing"), 'settings', 'eager_search',
            populate=False)
```

I haven't seen any other line of code that can change value of eager_search, but maybe I'm missing something.

**Reply by casdeiro** [reply]
It doesn't work either :-(

Next day I opened QL and `eager_search` is again set to `true` (how? by which line of code?!?)

This is absolutely frustrating and not being able to search in the covers view without the program freezing for several seconds is very irritating indeed!

**Reply by casdeiro** [reply]
Hi, people!

Finally I found a way to get rid of the `eager_search=true`. The problem of it going back to `true` every time after I having changed it to false was due to **QL overwriting it while running**. No matter if you change it through the GUI or if you manually change the value in the `config` file. The program, I think everytime the song changed, overwrote the contents of the config file. And it does it with the **values loaded on start**. So, the trick was to change the config file to reflect that `eager_search=false` value, **before starting QuodLibet!**

In that way, the loaded values and overwritten values will have the original value of `false` unchanged. And if you want to activate it back, you should do the same: change it after closing the program.

I've tested it with some start and close cycles of QL and even restarting the computer, and `eager_search=false` is still preserved! 

This might give a clue for the developer about the reason for the selection of the kind of search not being kept when it's done via the GUI, and a way for other people to get rid once and for all of the annoying "eager search".

Hope this really helps this time! :-D
